plugin/ocgrpc/stats_common.go

```diff
@@ -142,18 +142,21 @@ func handleRPCEnd(ctx context.Context, s *stats.End) {
 
 	latencyMillis := float64(elapsedTime) / float64(time.Millisecond)
 	if s.Client {
-		ctx, _ = tag.New(ctx,
-			tag.Upsert(KeyClientMethod, methodName(d.method)),
-			tag.Upsert(KeyClientStatus, st))
-		ocstats.Record(ctx,
+		ocstats.RecordWithTags(ctx,
```

Comment:
Not sure if the change helps these cases. RecordWithTags is going to return without recording if there is an error.